packages/devtools_app/lib/src/trees.dart

```diff
@@ -90,6 +90,9 @@ class TreeNode<T extends TreeNode<T>> {
     _isExpanded = true;
   }
 
+  // TODO(jacobr): cache the value of whether the node should be shown
+  // so that lookups on this class are O(1) invalidating the cache when nodes
+  // up the tree are expanded and collapsed.
```

Comment:
Do we want to invalidate the cache in this situation? If a parent with expanded children is collapsed, don't we want to preserve the expanded state of it's children so that, once the parent is expanded again, the children resume their expanded state?

Reply:
We would because the TreeTable might still be displaying those nodes animating them away so needs to know that they are changing to a hidden state.+